cmd/frontend/graphqlbackend/user_usage_stats.go

```diff
@@ -126,6 +128,9 @@ func (r *schemaResolver) LogEvents(ctx context.Context, args *EventBatch) (*Empt
 		return payload, nil
 	}
 
+	userID := actor.FromContext(ctx).UID
+	userPrimaryEmail, _, _ := r.db.UserEmails().GetPrimaryEmail(ctx, userID)
```

Comment:
If this is related to .com only, let's make a DB query only in this case too. Otherwise, we will make a bunch of useless DB queries on all other instances:
```suggestion
    userPrimaryEmail := ""
	if envvar.SourcegraphDotComMode() {
	    userPrimaryEmail, _, _ = r.db.UserEmails().GetPrimaryEmail(ctx, userID)
    }
```

Reply:
Great call!